### ISSUE TITLE
Clarification on learning rate scheduler settings for reproduction

### ISSUE THREAD

**Issue by Hollyhou423** [princeton-pli/retaining-by-doing#6]
Hi,
Thank you very much for releasing the paper and code. I am trying to reproduce the training setup and had a small question regarding the learning rate scheduler configuration.
In the paper, the training details state:
> “We used the AdamW optimizer. The learning rate was initialized to (1e^{-4}) for Llama-3.2-1B-Instruct and Qwen-2.5-1.5B-Instruct and (5e^{-6}) for Llama-3.1-8B-Instruct and Qwen2.5-7B-Instruct. We use cosine scheduler with warm-up step ratio 0.03 over the course of training.”
However, in the released code/configuration, I noticed that the scheduler-related settings appear to be:

```python
warmup_ratio = 0.0
scheduler_type = "constant"
```

Could you please clarify which setting should be used for faithful reproduction of the reported results?

Specifically, should we follow the paper and use:

```python
warmup_ratio = 0.03
scheduler_type = "cosine"
```

or should we follow the released code and use:

```python
warmup_ratio = 0.0
scheduler_type = "constant"
```

Thank you again for your work and for making the implementation available. I would really appreciate any clarification.
